src/hyperfine/benchmark.rs

```diff
@@ -221,17 +231,21 @@ pub fn run_benchmark(
                 options.failure_action,
                 None,
             )?;
-            progress_bar.inc(1);
+            if let Some(pbar) = progress_bar.clone() { pbar.inc(1); }
```

Comment:
We should avoid `.clone()`ing. You can probably use `if let Some(ref pbar) = …`

Reply:
Yeah, thanks, you were right, that works well and improves performance, although that (performance) doesn't seem crucial for this case?

Reply:
no, it's not. but I think its much cleaner without cloning it everywhere.

Reply:
yes, totally agree on that

Reply:
Coming to think of it, there is an even cleaner solution to write this, that we should probably follow. You can use `map` to "directly" call a function on the object inside the `Option`, allowing us to get rid of all the `if` statements. Should be something like
``` rust
progress_bar.map(|bar| bar.inc(1));
```
or
``` rust
progress_bar.as_ref().map(|bar| bar.inc(1));
```

Reply:
Hmm, that looks quite interesting, but do you think it's that readable?

Reply:
Yes, I think that's readable. Using `.map` (on `Option`, `Result`, or any kind of `Iterator`) is very common in Rust code.

Reply:
Actually, seems quite nice when I researched a bit more. Updated the PR with that way